### ISSUE TITLE
Code Completion and Intellisense Issue has regressed

### ISSUE THREAD

**Issue by ramvdixit** [flutter/flutter-intellij#6023]
The issue with code completion and intellisense is back. Here is my flutter doctor report! Code completion is mostly is  broken with suggestions here and there and auto suggestions for class imports fail. Also unused variables and objects are not highlighted (greyed-out). I have updated the Dart and Flutter plugins from the Preferences > Plugins

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel beta, 2.11.0-0.1.pre, on macOS 12.1 21C52 darwin-x64, locale en-IN)
    • Flutter version 2.11.0-0.1.pre at /Users/maheshiyer/Documents/flutter_sdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b101bfe32f (3 weeks ago), 2022-02-16 07:36:54 -0800
    • Engine revision e355993572
    • Dart version 2.17.0 (build 2.17.0-69.2.beta)
    • DevTools version 2.10.0-dev.1

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/maheshiyer/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/maheshiyer/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)

[✓] Connected device (3 available)
    • HD1911 (mobile) • e45b2c96 • android-arm64  • Android 11 (API 30)
    • macOS (desktop) • macos    • darwin-x64     • macOS 12.1 21C52 darwin-x64
    • Chrome (web)    • chrome   • web-javascript • Google Chrome 99.0.4844.51

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

**Comment by jacob314** [comment]
Fyi @jwren

**Comment by jwren** [comment]
tldr; You should be able to (re-)install the Dart and Flutter for IntelliJ to get the functionality to work again.

longer version:

A corrupt version of the Dart Plugin was pushed temporarily to the JetBrains servers for version `2021.1`, the issue was caught and fixed.  Please let me know if you are unsuccessful in installing the Dart and Flutter plugin.

**Comment by Sahkay** [comment]
I am having almost the exact same issues after updating my android studio, flutter/dart plugins and flutter version to the .3 hotfix. I did the updates after the previous comment stating the issue was a corrupt plugin. Reinstalling the flutter and dart plugins did not fix the issue either.

flutter doctor -v
`[✓] Flutter (Channel stable, 2.10.3, on macOS 11.6 20G165 darwin-x64, locale en-AU)
    • Flutter version 2.10.3 at /Users/joshuagil/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7e9793dee1 (5 days ago), 2022-03-02 11:23:12 -0600
    • Engine revision bd539267b4
    • Dart version 2.16.1
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/joshuagil/Library/Android/sdk
    • Platform android-31, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)

[✓] VS Code (version 1.65.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.36.0

[✓] Connected device (2 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86    • Android 11 (API 30) (emulator)
    • Chrome (web)                       • chrome        • web-javascript • Google Chrome 99.0.4844.51

[✓] HTTP Host Availability
    • All required HTTP hosts are available`

IDE info:

Android Studio Bumblebee | 2021.1.1 Patch 2
Build #AI-211.7628.21.2111.8193401, built on February 17, 2022
Runtime version: 11.0.11+0-b60-7590822 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 11.6
GC: G1 Young Generation, G1 Old Generation
Memory: 3000M
Cores: 12
Registry: external.system.auto.import.disabled=true, ide.suppress.double.click.handler=true
Non-Bundled Plugins: com.herbert.george.flutter-snippets (2.0.0-stable-1), Dart (211.7808), org.jetbrains.kotlin (211-1.6.10-release-923-AS7442.40), io.flutter (65.1.2), de.mariushoefler.flutter_enhancement_suite (1.6.0), com.squareup.sqldelight (1.5.3), pl.pszklarska.pubversionchecker (1.3.3)

**Comment by ramvdixit** [comment]
@stevemessick , @jacob314 @jwren 

I switched to Stable channel and ran flutter upgrade. Still the same issue. Here is my Doctor report:

[✓] Flutter (Channel stable, 2.10.3, on macOS 12.1 21C52 darwin-x64, locale en-IN)
    • Flutter version 2.10.3 at /Users/maheshiyer/Documents/flutter_sdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7e9793dee1 (6 days ago), 2022-03-02 11:23:12 -0600
    • Engine revision bd539267b4
    • Dart version 2.16.1
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/maheshiyer/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/maheshiyer/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)

[✓] VS Code (version 1.65.0)
    • VS Code at /Users/maheshiyer/Downloads/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (3 available)
    • HD1911 (mobile) • e45b2c96 • android-arm64  • Android 11 (API 30)
    • macOS (desktop) • macos    • darwin-x64     • macOS 12.1 21C52 darwin-x64
    • Chrome (web)    • chrome   • web-javascript • Google Chrome 99.0.4844.51

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

**Comment by ramvdixit** [comment]
Switching to VS Code until this is resolved. It's quite a task without the auto-suggestions and code-completion. Hoping that this issue will be resolved asap!

Thanks.

**Comment by jwren** [comment]
Using this Flutter Sdk version `2.10.3`, the Dart Plugin (version `211.7808`), the Flutter Plugin (version `65.2.2`), I am not reproducing this issue.

If the IntelliJ plugins are installed and enabled and you are still not able to get analysis, please attach the IntelliJ logs/ errors to assist in us troubleshooting. https://intellij-support.jetbrains.com/hc/en-us/articles/207241085-Locating-IDE-log-files

**Comment by Sahkay** [comment]
I updated to the latest version and after inspecting the logs I have found the below concerning error. This seems to be the only error that triggered directly after restarting the ide to install the latest version:

`ERROR - til.concurrency.QueueProcessor - null 
java.lang.NullPointerException
	at com.intellij.build.progress.BuildProgressImpl.getBuildId(BuildProgressImpl.java:31)
	at com.intellij.build.progress.BuildRootProgressImpl.getId(BuildRootProgressImpl.java:26)
	at com.intellij.build.progress.BuildProgressImpl.message(BuildProgressImpl.java:120)
	at com.intellij.build.progress.BuildRootProgressImpl.message(BuildRootProgressImpl.java:14)
	at com.intellij.compiler.progress.BuildOutputService.addMessage(BuildOutputService.java:176)
	at com.intellij.compiler.progress.CompilerTask.addMessage(CompilerTask.java:247)
	at com.intellij.compiler.impl.CompileContextImpl.addToProblemsView(CompileContextImpl.java:120)
	at com.intellij.compiler.impl.CompileContextImpl.addMessage(CompileContextImpl.java:105)
	at com.intellij.openapi.compiler.CompileContext.addMessage(CompileContext.java:63)
	at com.intellij.openapi.compiler.CompileContext.addMessage(CompileContext.java:46)
	at org.jetbrains.android.compiler.AndroidAutogenerator$1.compute(AndroidAutogenerator.java:133)
	at org.jetbrains.android.compiler.AndroidAutogenerator$1.compute(AndroidAutogenerator.java:117)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:840)
	at org.jetbrains.android.compiler.AndroidAutogenerator.runBuildConfigGenerator(AndroidAutogenerator.java:116)
	at org.jetbrains.android.compiler.AndroidAutogenerator.run(AndroidAutogenerator.java:94)
	at org.jetbrains.android.CreateAlarmForAutoGenerationAndroidStartupActivity.doGenerate(CreateAlarmForAutoGenerationAndroidStartupActivity.java:143)
	at org.jetbrains.android.CreateAlarmForAutoGenerationAndroidStartupActivity.generate(CreateAlarmForAutoGenerationAndroidStartupActivity.java:119)
	at org.jetbrains.android.CreateAlarmForAutoGenerationAndroidStartupActivity.access$000(CreateAlarmForAutoGenerationAndroidStartupActivity.java:48)
	at org.jetbrains.android.CreateAlarmForAutoGenerationAndroidStartupActivity$1.run(CreateAlarmForAutoGenerationAndroidStartupActivity.java:69)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:238)
	at com.intellij.util.Alarm$Request.runSafely(Alarm.java:369)
	at com.intellij.util.Alarm$Request.run(Alarm.java:355)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.intellij.util.concurrency.SchedulingWrapper$MyScheduledFutureTask.run(SchedulingWrapper.java:220)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:216)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$200(BoundedTaskExecutor.java:27)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.execute(BoundedTaskExecutor.java:195)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:213)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:184)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:829)
2022-03-09 11:11:12,216 [  16929]  ERROR - til.concurrency.QueueProcessor - Android Studio Bumblebee | 2021.1.1 Patch 2  Build #AI-211.7628.21.2111.8193401 
2022-03-09 11:11:12,216 [  16929]  ERROR - til.concurrency.QueueProcessor - JDK: 11.0.11; VM: OpenJDK 64-Bit Server VM; Vendor: JetBrains s.r.o. 
2022-03-09 11:11:12,217 [  16930]  ERROR - til.concurrency.QueueProcessor - OS: Mac OS X 
2022-03-09 11:11:12,218 [  16931]  ERROR - til.concurrency.QueueProcessor - Last Action:  
2022-03-09 11:11:18,725 [  23438]   INFO - ter.jxbrowser.JxBrowserManager - rafflenumbers: JxBrowser file downloaded: /Users/joshuagil/Library/Application Support/Google/AndroidStudio2021.1/plugins/flutter-intellij/jxbrowser/jxbrowser-mac-7.22.jar`

**Comment by ramvdixit** [comment]
Here is my IntelliJ log:

The title of "IDE Fatal Errors" dialog is: '1 of 1 Exception in plugin Flutter (65.1.2).'

java.lang.IllegalStateException: Module name 'hushlock_android' already exists. Trying to load module: /Users/maheshiyer/Documents/Workspace/Product/hushlock/android/hushlock_android.iml
	at com.intellij.workspaceModel.ide.impl.legacyBridge.module.ModifiableModuleModelBridgeImpl.loadModule(ModifiableModuleModelBridgeImpl.kt:156)
	at io.flutter.module.FlutterModuleBuilder.addAndroidModule(FlutterModuleBuilder.java:211)
	at io.flutter.FlutterInitializer.runActivity(FlutterInitializer.java:150)
	at com.intellij.ide.startup.impl.StartupManagerImpl.runStartupActivity(StartupManagerImpl.java:335)
	at com.intellij.ide.startup.impl.StartupManagerImpl.runActivity(StartupManagerImpl.java:309)
	at com.intellij.ide.startup.impl.StartupManagerImpl.lambda$runPostStartupActivities$3(StartupManagerImpl.java:245)
	at com.intellij.openapi.project.DumbServiceImpl.doRun(DumbServiceImpl.java:403)
	at com.intellij.openapi.project.DumbServiceImpl.updateFinished(DumbServiceImpl.java:395)
	at com.intellij.openapi.project.TrackedEdtActivityService$TrackedEdtActivity.run(TrackedEdtActivityService.java:66)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:218)
	at com.intellij.openapi.application.TransactionGuardImpl.access$200(TransactionGuardImpl.java:21)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:200)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:787)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$invokeLater$4(ApplicationImpl.java:325)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:84)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:133)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:46)
	at com.intellij.openapi.application.impl.FlushQueue$FlushNow.run(FlushQueue.java:189)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:969)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:839)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:449)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:808)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$9(IdeEventQueue.java:448)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:787)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:496)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)

**Comment by jwren** [comment]
Last comment -- this is a dup of https://github.com/flutter/flutter-intellij/issues/6019, an additional Flutter Plugin minor version was pushed out to resolve this issue.

**Comment by stevemessick** [comment]
@ramvdixit You need to get the latest version of the plugin, 65.2.2. You have 65.1.2, which is broken.
